efi/env.go

```diff
@@ -37,8 +38,10 @@ import (
 // consumers of the API can provide a custom mechanism to read EFI variables or parse
 // the TCG event log.
 type HostEnvironment interface {
-	// ReadVar reads the specified EFI variable
-	ReadVar(name string, guid efi.GUID) ([]byte, efi.VariableAttributes, error)
+	// VarContext returns a context containing a VarsBackend, keyed by efi.VarsBackendKey,
+	// for interacting with EFI variables via go-efilib. This context can be passed to any
+	// go-efilib function that interacts with EFI variables.
+	VarContext() context.Context
```

Comment:
this is a bit a strange API for contexts, if a caller gets a context ti usually should pass it along, and not start with a fresh one, so a more expected API would be:

```
VartContext(parent context.Context) context.Context
```

which would return something derived from parent if parent is not nil, instead of returning a fresh context in all cases

Reply:
Funnily enough, I'd already added an API for this, so I've just pulled in the go-efilib version with it and used it in the default implementation.